src/vcpkg/metrics.cpp

```diff
@@ -608,32 +516,105 @@ namespace vcpkg
         fs.write_contents(vcpkg_metrics_txt_path, payload, ec);
         if (ec) return;
 
-#if defined(_WIN32)
-        const Path temp_folder_path_exe = temp_folder_path / "vcpkg-" VCPKG_BASE_VERSION_AS_STRING ".exe";
+        const StringLiteral executableExtension =
+#if defined(WIN32)
+            ".exe"
+#else
+            ""
+#endif
+            ;
+
+        const Path temp_folder_path_exe =
+            temp_folder_path / "vcpkg-" VCPKG_BASE_VERSION_AS_STRING + executableExtension;
```

Comment:
```suggestion
        const Path temp_folder_path_exe =
            temp_folder_path / "vcpkg-" VCPKG_BASE_VERSION_AS_STRING
            
#if defined(WIN32)
            ".exe"
#endif
            ;
```

We're already using string combining so we may as well use it to add the `.exe` too